media_driver/agnostic/common/codec/hal/codechal_vdenc_hevc.cpp

```diff
@@ -796,14 +799,18 @@ MOS_STATUS CodechalVdencHevcState::SetupMbQpStreamIn(PMOS_RESOURCE streamIn)
             //            uint32_t YOffset = (h % 2) * 2;
             //            uint32_t XOffset = 2 * (w/2 * 2) + w % 2;
 
-            int32_t ForceQp = pInputData[(h/2) * m_encodeParams.psMbQpDataSurface->dwPitch + (w/2)];
             //            (pData + (Offset + XOffset + YOffset))->DW7.QpEnable = 0xf;
             //            (pData + (Offset + XOffset + YOffset))->DW14.ForceQp_0 = ForceQp;
             //            (pData + (Offset + XOffset + YOffset))->DW14.ForceQp_1 = ForceQp;
             //            (pData + (Offset + XOffset + YOffset))->DW14.ForceQp_2 = ForceQp;
             //            (pData + (Offset + XOffset + YOffset))->DW14.ForceQp_3 = ForceQp;
             streaminDataParams.setQpRoiCtrl = true;
-            streaminDataParams.forceQp = (int8_t)ForceQp;
+
+            streaminDataParams.forceQp[0] = (int8_t) ( pInputData[(h * 2)     * m_encodeParams.psMbQpDataSurface->dwPitch + (w * 2)]);
```

Comment:
Hi Xiaoxia,
May I know what's the block size of input MbQpDataSurface?
Is the MB QP map in 16x16 block or 32x32 block or other size?

Thanks,
Thomas

Reply:
I'm asking because this change may be conflict with HW design and validation results. StreamIn surface 4 QP values need to be the same according to bspec.

Reply:
> Hi Xiaoxia,
> May I know what's the block size of input MbQpDataSurface?
> Is the MB QP map in 16x16 block or 32x32 block or other size?
> 
> Thanks,
> Thomas

Hi Thomas:
The block size is 16x16 for HEVC VDENC, which is defined in MSDK. As we tested here on SG1, it can work as expected. Each 16x16 block can be set to specified QP value. Is it possible the HW design limitation has already been removed or resolved for Gen12?